src/modules/peek/Peek.FilePreviewer/Previewers/MediaPreviewer/VideoPreviewer.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using CommunityToolkit.Mvvm.ComponentModel;
+using Microsoft.PowerToys.Settings.UI.Library;
+using Microsoft.UI.Dispatching;
+using Peek.Common.Extensions;
+using Peek.Common.Helpers;
+using Peek.Common.Models;
+using Peek.FilePreviewer.Previewers.Interfaces;
+using Windows.Foundation;
+using Windows.Media.Core;
+using Windows.Storage;
+
+namespace Peek.FilePreviewer.Previewers
+{
+    public partial class VideoPreviewer : ObservableObject, IVideoPreviewer, IDisposable
+    {
+        [ObservableProperty]
+        private MediaSource? preview;
+
+        [ObservableProperty]
+        private PreviewState state;
+
+        [ObservableProperty]
+        private Size videoSize;
+
+        public VideoPreviewer(IFileSystemItem file)
+        {
+            Item = file;
+            Dispatcher = DispatcherQueue.GetForCurrentThread();
+        }
+
+        private IFileSystemItem Item { get; }
+
+        private DispatcherQueue Dispatcher { get; }
+
+        private Task<bool>? VideoTask { get; set; }
+
+        /*private bool IsVideoLoaded => VideoTask?.Status == TaskStatus.RanToCompletion;*/
```

Comment:
Will it be used later? I believe it'd be good to leave a comment or remove it.

Reply:
Removed.